src/EntityFramework/Core/EntityClient/DbParameterCollectionHelper.cs

```diff
@@ -292,6 +292,50 @@ private static int IndexOf(IEnumerable items, string parameterName)
             return -1;
         }
 
+        /// <inhertidoc />
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1604:ElementDocumentationMustHaveSummary")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1611:ElementParametersMustBeDocumented")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1615:ElementReturnValueMustBeDocumented")]
+        [SuppressMessage("Microsoft.Usage", "CA2201:DoNotRaiseReservedExceptionTypes")]
+        public DbParameter GetParameterByValue(string value)
+        {
+            var index = IndexOfParameterValue(InnerList, value);
+            if (index < 0)
+            {
+                return null;
+            }
+
+            return InnerList[index];
+        }
+
+        private static int IndexOfParameterValue(IEnumerable items, string parameterValue)
```

Comment:
I'd be happy that the fix would still be effective if the second check was removed. In the vast majority of cases the values would be identical or not match at all.
Now thinking about it from a different angle, it's probably more efficient to drop the second compare. At best it will only provide a match a handful of times, yet will run through the collection a second time for every new parameter. I was so focussed on reducing the number of parameters it was generating I missed this.
I'd agree that the second check isn't warranted, created noise, and should be removed. Thanks for spotting it.